src/components/Filter/Filter.jsx

```diff
@@ -1,14 +1,73 @@
-import React from 'react';
+import React, {useState, useCallback} from 'react';
+import { connect } from 'react-redux'
+import PropTypes from 'prop-types';
+import useToggle from '../../hooks/useToggle.jsx'
+import Menu from '../Common/Menu';
+import {movieActions} from '../../store/actions';
 import './Filter.css';
 
-const Filter = (props) => (
-    <ul className='filterList'>
-        <li className='filterListItem active'>All</li>
-        <li className='filterListItem'>Documentary</li>
-        <li className='filterListItem'>Comedy</li>
-        <li className='filterListItem'>Horror</li>
-        <li className='filterListItem'>Crime</li>
-    </ul>
-);
-
-export default Filter;
+const Filter = ({genres, setGenreForFilter, genreForFilter}) => {
+    const [isOpened, toggleIsOpened] = useToggle();
+
+    const modifiedGenres = genres.map((genre) => {
+        return {
+            id: genre,
+            title: genre,
+            onClick: () => {
+                setGenreForFilter(genre);
+                toggleIsOpened();
+            },
+        };
+    });
+
+    const genresList = (
+        <ul className='filterList'>
+            {genres.map((genre) => (
+                <li
+                    key={genre}
+                    className={`filterListItem ${
+                        genre == genreForFilter ? 'active' : ''
+                    }`}
+                    onClick={() => setGenreForFilter(genre)}
+                >
+                    {genre}
+                </li>
+            ))}
+        </ul>
+    );
+
+    const genresMenu = (
+        <div className='filter'>
+            <div className='filterLabel'>Filter by genre</div>
+            <div className='filterMenu'>{genreForFilter}</div>
+            <button className='caretButton' onClick={() => toggleIsOpened()}>
+                <span className={isOpened ? 'caret-up' : 'caret-down'}></span>
+            </button>
+            {isOpened && <Menu menuItems={modifiedGenres} />}
+        </div>
+    );
+
+    return (
+        <>
+            {genres.length <= 6 && genresList}
```

Comment:
why 6?)

Reply:
I decided so) 6 genres fit quite well in a line in my application, more will be presented in the form of a dropdown.